tests/e2e/Payment.test.ts

```diff
@@ -317,17 +340,38 @@ describe('e2e', () => {
           await wait()
           expect(txHash).to.be.a('string')
         })
+
+        it('should confirm trustline transfer with message', async () => {
+          const preparedPayment = await tl1.payment.prepare(
+            network.address,
+            user2.address,
+            1
+          )
+          const message = 'testMessage'
+          const txHash = await tl1.payment.confirmPayment(
+            preparedPayment,
+            message
+          )
+          await wait()
+          expect(txHash).to.be.a('string')
+        })
       })
 
       describe('#get()', () => {
+        let messageId: string
+
         before(async () => {
-          const { rawTx } = await tl1.payment.prepare(
+          const preparedPayment = await tl1.payment.prepare(
             network.address,
             user2.address,
             1.5,
-            { paymentRequestId }
+            {
+              paymentRequestId,
+              addMessageId: true
+            }
           )
-          await tl1.payment.confirm(rawTx)
+          messageId = preparedPayment.messageId
+          await tl1.payment.confirm(preparedPayment.rawTx)
```

Comment:
Yes, I don't really test for the message, so it's superfluous to add a message here.
But I am unsure about deprecating `confirm`, my intention was to first deprecate `confirm` because otherwise the app team will fail to change it and have bugs. But also at a later point deprecate `confirmPayment` and replace `confirm` by `confirmPayment`.

Maybe that's not the best plan.

Reply:
Since the parameter count differs, we may could consider to use function overloading here? So we have the `confirm` function twice and at some point we remove the depicted one without the trouble of renaming. This procedure should work as long as we don't want to replace a function with the same amount of parameter but change the type. This should not work since the transcompiled JavaScript can't differ them.

Reply:
Just as note here: we discussed in a private chat that overloading in TypeScript is awful.